src/clr-angular/utils/focus/focus-on-view-init/focus-on-view-init.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { AfterViewInit, Directive, ElementRef, HostListener, Inject, Input, PLATFORM_ID } from '@angular/core';
+import { isPlatformBrowser } from '@angular/common';
+
+/* This directive is for guiding the document focus to the new added content when its view is initialized. */
+@Directive({
+  selector: '[clrFocusOnViewInit]',
+  host: {
+    '[attr.tabindex]': 'tabIndex',
+  },
+})
+export class ClrFocusOnViewInit implements AfterViewInit {
+  constructor(private el: ElementRef, @Inject(PLATFORM_ID) private platformId: Object) {}
+
+  // negative tabindex as we shouldn't have this in the tab order
+  private _tabIndex: number = -1;
+
+  // set to true as we will intend to set the tab index by default
+  // if the element the directive is placed on has tabindex already, we will use that existing one.
+  private tabIndexSet: boolean = true;
```

Comment:
This will always be set. Does it sound more natural to call it `hasDefaultTabIndex` or `hasCustomTabIndex`?

Reply:
I like `hasDefaultTabIndex`. I will use it. Thanks for the suggestion.